cms/djangoapps/contentstore/views/course.py

```diff
@@ -542,43 +547,72 @@ def format_in_process_course_view(uca):
             ) if uca.state == CourseRerunUIStateManager.State.FAILED else u''
         }
 
-    def format_library_for_view(library):
-        """
-        Return a dict of the data which the view requires for each library
-        """
-
-        return {
-            u'display_name': library.display_name,
-            u'library_key': six.text_type(library.location.library_key),
-            u'url': reverse_library_url(u'library_handler', six.text_type(library.location.library_key)),
-            u'org': library.display_org_with_default,
-            u'number': library.display_number_with_default,
-            u'can_edit': has_studio_write_access(request.user, library.location.library_key),
-        }
-
     split_archived = settings.FEATURES.get(u'ENABLE_SEPARATE_ARCHIVED_COURSES', False)
     active_courses, archived_courses = _process_courses_list(courses_iter, in_process_course_actions, split_archived)
     in_process_course_actions = [format_in_process_course_view(uca) for uca in in_process_course_actions]
 
     return render_to_response(u'index.html', {
-        u'courses': active_courses,
-        u'archived_courses': archived_courses,
-        u'in_process_course_actions': in_process_course_actions,
-        u'libraries_enabled': LIBRARIES_ENABLED,
-        u'redirect_to_library_authoring_mfe': should_redirect_to_library_authoring_mfe(),
-        u'library_authoring_mfe_url': LIBRARY_AUTHORING_MICROFRONTEND_URL,
-        u'libraries': [format_library_for_view(lib) for lib in libraries],
-        u'show_new_library_button': get_library_creator_status(user) and not should_redirect_to_library_authoring_mfe(),
-        u'user': user,
-        u'request_course_creator_url': reverse('request_course_creator'),
-        u'course_creator_status': _get_course_creator_status(user),
-        u'rerun_creator_status': GlobalStaff().has_user(user),
-        u'allow_unicode_course_id': settings.FEATURES.get(u'ALLOW_UNICODE_COURSE_ID', False),
-        u'allow_course_reruns': settings.FEATURES.get(u'ALLOW_COURSE_RERUNS', True),
-        u'optimization_enabled': optimization_enabled
+        'courses': active_courses,
+        'split_studio_home': split_library_view_on_dashboard(),
+        'archived_courses': archived_courses,
+        'in_process_course_actions': in_process_course_actions,
+        'libraries_enabled': LIBRARIES_ENABLED,
+        'redirect_to_library_authoring_mfe': should_redirect_to_library_authoring_mfe(),
+        'library_authoring_mfe_url': LIBRARY_AUTHORING_MICROFRONTEND_URL,
+        'libraries': [_format_library_for_view(lib, request) for lib in libraries],
```

Comment:
library authoring MFE is used to edit or create library and is not relevant to this line, and libraries are formated here to list it on the dashboard in case the switch is not enabled. FYI:  I just removed `u''` from this line and this is legacy implementation.

Reply:
I think I am missing something here. 
As I can read this, I see libraries are added/formatted in both cases either switch is on or off. 

If you have some time on we can sort this out on call.